apis/core/v1alpha2/core_scope_types.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// A HealthScopeSpec defines the desired state of a HealthScope.
+type HealthScopeSpec struct {
+	// ProbeTimeout is the amount of time in seconds to wait when receiving a response before marked failure.
+	ProbeTimeout *int32 `json:"probe-timeout,omitempty"`
+
+	// ProbeInterval is the amount of time in seconds between probing tries.
+	ProbeInterval *int32 `json:"probe-interval,omitempty"`
+
+	// FailurePercentageThreshold is the minimum percentage of failure of total probe results for health to be declared 'failed'.
+	FailureRateThreshold *int32 `json:"failure-rate-threshold,omitempty"`
+
+	// HealthyPercentageThreshold is the minimum percentage of healthy of total probe results for health to be declared 'healthy'.
+	HealthyRateThreshold *int32 `json:"healthy-rate-threshold,omitempty"`
+
+	// WorkloadReferences to the workloads that are in this scope.
+	WorkloadReferences []runtimev1alpha1.TypedReference `json:"workloadRefs,omitempty"`
```

Comment:
Shouldn't this be in the status?
Components are added to a HealthScope after HealthScope is created.

Reply:
Great point, I can move to status.

Reply:
Done. We still need to change workloadRef in trait to make it consistent.